### PR TITLE
Add note about Scala 2.12 default to 2.4.2 release notes

### PR DESCRIPTION
**Pull request by srowen**

### CODE REVIEW

File: site/releases/spark-release-2-4-2.html [hunk 1]
```diff
@@ -205,10 +205,13 @@ <h2>Spark Release 2.4.2</h2>
 
 <p>Spark 2.4.2 is a maintenance release containing stability fixes. This release is based on the branch-2.4 maintenance branch of Spark. We strongly recommend all 2.4 users to upgrade to this stable release.</p>
 
+<p>Note that Scala 2.11 support was deprecated in 2.4.1, and as of 2.4.2, the pre-built convenenience binaries are compiled for Scala 2.12.
```
Comment (gaborgsomogyi) [inline site/releases/spark-release-2-4-2.html]: Maybe 2.4.1+?

Comment (srowen) [reply site/releases/spark-release-2-4-2.html]: You mean "deprecated as of 2.4.1 and beyond"? Ok that's fine.

Comment (gaborgsomogyi) [reply site/releases/spark-release-2-4-2.html]: Yeah.

**Review by cloud-fan — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by gaborgsomogyi — APPROVED** [review]
The change itself looks good but from the mail discussion it's not clear to me whether this is a bug or a feature.

**Comment by srowen** [comment]
Agree, I figure it's best to call extra attention to it for now, either way.

**Pull request closed without merging**